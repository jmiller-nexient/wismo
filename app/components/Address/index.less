@import '../../styles/themes/index.less';

@address-prefix-cls: ~'@{dte-prefix}-address';

.@{address-prefix-cls} {
  color: @dte-gray-600;
  font-family: @dte-wismo-font-family;
  font-style: @dte-wismo-font-style-normal;
  line-height: 1.375rem;
}
