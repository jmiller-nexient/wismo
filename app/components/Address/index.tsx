import React from 'react';

import './index.less';

interface IAddressProps {
  readonly city: string;
  readonly line1: string;
  readonly line2?: string | null;
  readonly state: string;
  readonly zip: string;
}

const Address: React.FC<IAddressProps> = (props: IAddressProps): React.ReactElement => {
  const { city, line1, line2, state, zip } = props;

  return (
    <address className="dte-wismo-address">
      {line1 + (line2 ? `, ${line2}` : '')}
      <br />
      {`${city}, ${state} ${zip}`}
    </address>
  );
};

export default Address;
