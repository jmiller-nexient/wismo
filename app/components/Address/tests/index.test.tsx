import React from 'react';
import renderer from 'react-test-renderer';

import Address from '..';

describe('Address Component', () => {
  it('should not render line 2 when null', () => {
    const address = {
      line1: '123 Main St',
      line2: null,
      city: 'Bloomfield',
      state: 'MI',
      zip: '48301',
    };

    const tree = renderer.create(
      <Address {...address} />,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should render line2 when it has a value', () => {
    const address = {
      line1: '123 Main St',
      line2: 'Apt. 213',
      city: 'Bloomfield',
      state: 'MI',
      zip: '48301',
    };

    const tree = renderer.create(
      <Address {...address} />,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
