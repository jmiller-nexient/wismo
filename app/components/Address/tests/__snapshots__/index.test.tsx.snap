// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Address Component should not render line 2 when null 1`] = `
<address
  className="dte-wismo-address"
>
  123 Main St
  <br />
  Bloomfield, MI 48301
</address>
`;

exports[`Address Component should render line2 when it has a value 1`] = `
<address
  className="dte-wismo-address"
>
  123 Main St, Apt. 213
  <br />
  Bloomfield, MI 48301
</address>
`;
